File: apps/backend/src/domain/product/ProductRepository.ts
import type { Product } from './Product.js'

export interface ProductRepository {
  findAll(): Promise<Product[]>
  findById(id: number): Promise<Product | null>
  save(product: Product): Promise<Product>
  update(id: number, product: Product): Promise<Product>
  delete(id: number): Promise<void>
}

File: apps/webshop/backend/src/main/java/com/frerememoire/webshop/domain/product/port/ProductRepository.java
package com.frerememoire.webshop.domain.product.port;

import com.frerememoire.webshop.domain.product.Product;

import java.util.List;
import java.util.Optional;

public interface ProductRepository {
    List<Product> findAll();
    List<Product> findAllActive();
    Optional<Product> findById(Long id);
    Product save(Product product);
    void deleteById(Long id);
    boolean existsById(Long id);
}
